### DISCUSSION TITLE
Rules issue / debug option?

### DISCUSSION THREAD

**Discussion by joergsx** [arendst/Tasmota#22413] [Support request]
Hi there,
I just can’t get the following rule to work. I would like to infer the status of my washing machine based on its power consumption. The washing machine has a power draw of 0W when turned off. In standby/finished mode, it draws 2W, and when in operation, the power is greater than 2W. When the power drops below 3W, the "standby" status should be published after a delay. There's some problem in the attached code that I can’t quite solve. Is there any way to debug this? I would be grateful for a small hint. I have already gone through the documentation with examples several times and tried various things.

`Var1 0`
`Rule1`
`ON ENERGY#Power>5 DO Backlog IF (%var1%!=1) var1 1; publish %topic%/wstatus läuft; ruletimer1 0 ENDIF BREAK`
`ON ENERGY#Power>1 DO Backlog IF (%var1%!=2) ruletimer1 60 ENDIF BREAK`
`ON ENERGY#Power<=1 DO Backlog IF (%var1%!=3) var1 3; publish %topic%/wstatus aus; ruletimer1 0 ENDIF BREAK`
`ON Rules#Timer=1 DO Backlog IF (%var1%!=2) var1 2; publish %topic%/wstatus fertig ENDIF endon`
`Rule1 1`

As a result, I get the error message quite frequently, which is also transmitted via mqtt. How can I switch off error transmission via mqtt to save load?

`20:49:33.539 RUL: ENERGY#POWER>1 performs "Backlog IF (0!=2) ruletimer1 60 ENDIF"`
`20:49:33.572 MQT: stat/basement/washing machine/COMMAND = {"Command":"Unknown"}`
`20:49:34.140 RUL: ENERGY#POWER>1 performs "Backlog IF (0!=2) ruletimer1 60 ENDIF"`
`20:49:34.174 MQT: stat/basement/washing machine/COMMAND = {"Command":"Unknown"}`
`20:49:34.741 RUL: ENERGY#POWER>1 performs "Backlog IF (0!=2) ruletimer1 60 ENDIF"`
`20:49:34.776 MQT: stat/basement/washing machine/COMMAND = {"Command":"Unknown"}`

Does anybody know my mistake or has any better solution for that issue?

Regards, thanky you

**Comment by barbudor** [comment]
I would bet you are not using a build supporting the If statement which is not included in default tasmota for esp8266

**Reply by sfromis** [reply]
For recent versions of Tasmota `if/endif` is included in the default builds, also for ESP8266, but I suppose that barbudor is right about you using an older version. I'd suggest to upgrade.

You could also verify with `Weblog 3`, after which the debug messages will indicate what command is unknown, but it is highly likely that it is indeed the `if`.

**Reply by joergsx** [reply]
Thank you for that. You are right! I have the Program Version | 13.0.0(tasmota) Build Date & Time 2023-08-16T10:19:59 Core/SDK Version 2_7_4_9/2.2.2-dev(38a443e) that was delivered with my NOUS A1T Smart Socket.
`Weblog 3` brings 
```
22:02:10.832 MQT: stat/keller/waschmaschine/COMMAND = {"Command":"Unknown"}
22:02:11.398 RUL: ENERGY#POWER>1 performs "Backlog IF (0!=2) ruletimer1 60 ENDIF"
22:02:11.399 SRC: Rule
22:02:11.402 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 29, Pld -99, Data 'IF (0!=2) ruletimer1 60 ENDIF'
22:02:11.425 SRC: Backlog
22:02:11.428 CMD: Grp 0, Cmd 'IF', Idx 1, Len 26, Pld -99, Data '(0!=2) ruletimer1 60 ENDIF'
22:02:11.438 MQT: stat/keller/waschmaschine/COMMAND = {"Command":"Unknown"}
```
